src/provider/ContextProvider.js

```diff
@@ -54,11 +55,15 @@ const ContextProvider = ({ children }) => {
       }
 
       const newCheckout = await createNewCheckout()
-      setCheckoutInState(newCheckout)
+      if (!isRemoved) {
+        setCheckoutInState(newCheckout)
+      }
     }
 
     initializeCheckout()
   }, [store.client.checkout])
+  
+  useEffect(() => () => { isRemoved = true; }, [])
```

Comment:
This has some code smell. See here: https://dev.to/juliang/cancelling-a-promise-with-react-useeffect-3062 then https://dev.to/sebastienlorber/handling-api-request-race-conditions-in-react-4j5b

Reply:
What do you mean, sorry? The first article explains my solution. Not sure what exactly you're trying to point out in the second article. Could you please elaborate?

Reply:
https://github.com/AlexanderProd/gatsby-shopify-starter/pull/43